### DISCUSSION TITLE
Deserializing JSON from an external API

### DISCUSSION THREAD

**Discussion by mozexty** [serverpod/serverpod#1914] [Q&A] [answered]
Let's say I've got this JSON from an external API.

`var responseBody = '[{"firstName":"John","lastName":"Doe","age":55},{"firstName":"Joe","lastName":"Harvey","age":12}]';
`
I've created the following person.spy.yaml file in order to deserialize it

class: Person
fields:
  firstName : String
  lastName : String
  age : int

How do can I get a `List<Person>` from that responseBody ?

**Comment by Isakdl** [comment]
Import the `Protocol` class and then call the `decode` method with the type you want to deserialize.

I.E.

```dart
import 'package:<project_name>_client/<project_name>_client.dart'; // If you are doing this in Flutter
import 'package:<project_name>_server/src/generated/protocol.dart'; // If you are doing this in Serverpod

var protocol = Protocol();

var person = protocol.decode<List<Person>>(responseBody);
```

**Reply by mozexty** [reply]
It says : FormatException: No deserialization found for type List<Person>

**Reply by Isakdl** [reply]
I see, we might only generate the parsing for List<Class> if it is declared as such in another model.

Then you have two options. Slightly hacky, but you could create another model that has your class as a List in a field. This will force the code generator to create the parsing and then the above code should work.

I.E.

```yaml
class: DummyPersonList
fields:
  persons: List<Person>
```

Second option is to decode the string your self and convert it, like so:

```dart
var protocol = Protocol();
var jsonObject = jsonDecode(responseBody);

var personList = (jsonObject as List).map((e) => Person.fromJson(e, protocol));
```

**Reply by mozexty** [reply, marked as answer]
Thank you @Isakdl 
The second option works.

But the first one doesn't. 
I added that DummyPersonList class, and I `serverpod generate`. Same exception as before. 
Then I took a look at the `T deserialize<T>(dynamic data, [Type? t]) {}` method in the generated serialization.dart, and I don't see any mention of the `List<T>` type.
